lib/3scale_toolbox/commands/remote_command/remote_add.rb

```diff
@@ -0,0 +1,75 @@
+require 'cri'
+require '3scale/api'
+require '3scale_toolbox/base_command'
+
+module ThreeScaleToolbox
+  module Commands
+    module RemoteCommand
+      class RemoteAddSubcommand < Cri::CommandRunner
+        include ThreeScaleToolbox::Command
+        def self.command
+          Cri::Command.define do
+            name        'add'
+            usage       'add <remote_name> <remote_url>'
+            summary     '3scale CLI remote add'
+            description '3scale CLI command to add new remote'
+            param       :remote_name
+            param       :remote_url
+            runner RemoteAddSubcommand
+          end
+        end
+
+        def run
+          # 'arguments' cannot be converted to Hash
+          add_remote arguments[:remote_name], arguments[:remote_url]
+        rescue StandardError => e
+          warn e.message
+          warn e.backtrace
+          exit 1
+        end
+
+        def validate_remote_name(name)
+          remotes = config.data :remotes
+          raise 'fatal: remote name already exists.' if !remotes.nil? && remotes.key?(name)
+        end
+
+        def parse_remote_uri(remote_url_str)
+          # should raise error on invalid urls
+          remote_uri_obj = URI(remote_url_str)
+          provider_key = remote_uri_obj.user
+          remote_uri_obj.user = ''
+          endpoint = remote_uri_obj.to_s
+          { provider_key: provider_key, endpoint: endpoint }
```

Comment:
It is not necessarily a provider_key. I'd just call it authentication.

Reply:
will call it `auth_key` because it is a key or token, not a method or process.